planning.py

```diff
@@ -94,24 +96,27 @@ def edge_is_in_collision(self, xy1, xy2, h=0.1):
             if self.point_is_in_collision(point):
                 return True
         return False
-
-    def sample(self, n=1, max_tries_factor=5):
+    def sample(self,graph ,n=1,step = 0.1):
         """Sample n points from the workspace that is collision-free.
 
         Will try generating up to max_tries_factor * n points in total, after
         which the samples are returned even if there are not n of them.
         """
-        samples = []
-        num_tries = 0
-        while len(samples) < n and num_tries < max_tries_factor * n:
+        samples=[]
+        for _ in range(n):
             # sample the entire workspace uniformly
             xy = (np.random.random(2) - 0.5) * [self.width, self.height]
 
             # add this point if it is collision-free
             if not self.point_is_in_collision(xy):
                 samples.append(xy)
+            else:
+                v,_ = graph.closest_vertex(xy)
+                v = v.coord
+                while self.point_is_in_collision(xy):
+                    xy = xy + (xy - v) * step / np.linalg.norm(xy - v)
```

Comment:
Do we need to worry about `xy = v` and a division by zero here?

Reply:
yes, I think you are right. That could be a problem